packages/ui/ImageItem/ImageItem.js

```diff
@@ -0,0 +1,243 @@
+
+		/**
+		 * Shows a selection overlay with a centered icon. When `selected` is true, a check mark is shown.
+		 *
+		 * @type {Boolean}
+		 * @default false
+		 * @public
+		 */
+		selectionOverlayShowing: PropTypes.bool,
+
+		/**
+		 * The absolute URL path to the image.
+		 *
+		 * @type {String}
+		 * @public
+		 */
+		source: PropTypes.string,
+
+		/**
+		 * The second caption line to be displayed with the image.
+		 *
+		 * @type {String}
+		 * @public
+		 */
+		subCaption: PropTypes.string,
+
+		/**
+		 * The components that will be shown near the image, aligned with `orientation`.
+		 *
+		 * @type {Array|Element}
+		 * @public
+		 */
+		subComponents: PropTypes.oneOfType([PropTypes.array, PropTypes.element])
+	},
+
+	defaultProps: {
+		captionComponent: 'div',
+		iconComponent: Icon,
+		imageComponent: Image,
+		orientation: 'vertical',
+		selected: false,
+		selectionOverlayShowing: false
+	},
+
+	styles: {
+		css: componentCss,
+		className: 'imageItem',
+		publicClassNames: true
+	},
+
+	computed: {
+		className: ({caption, selected, styler, subCaption}) => styler.append(
+			{selected, caption, subCaption}
+		),
+		selectionOverlay: ({css, iconComponent: IconComponent, selectionOverlay: SelectionOverlay, selectionOverlayShowing}) => {
+			if (selectionOverlayShowing) {
+				return (
+					<div className={css.overlayContainer}>
+						{
+							SelectionOverlay ?
+								<SelectionOverlay /> :
+								<div className={css.overlayComponent}>
+									<IconComponent className={css.icon}>check</IconComponent>
+								</div>
+						}
+					</div>
+				);
+			}
+		},
+		subComponents: ({caption, captionComponent: Caption, css, subCaption, subComponents}) => {
+			return (
+				subComponents ? subComponents : <React.Fragment>
+					{caption ? (<Cell className={css.caption} component={Caption} shrink>{caption}</Cell>) : null}
+					{subCaption ? (<Cell className={css.subCaption} component={Caption} shrink>{subCaption}</Cell>) : null}
+				</React.Fragment>
+			);
+		}
+	},
+
+	render: ({className, css, imageComponent: ImageComponent, orientation, placeholder, source, selectionOverlay, subComponents, ...rest}) => {
+		const classes = classNames(className, orientation === 'vertical' ? css.vertical : css.horizontal);
+
+		delete rest.caption;
+		delete rest.captionComponent;
+		delete rest.iconComponent;
+		delete rest.selected;
+		delete rest.selectionOverlayShowing;
+		delete rest.subCaption;
+
+		return (
+			orientation === 'vertical' ? <Column {...rest} className={classes} inline>
+				<Cell className={css.image} component={ImageComponent} placeholder={placeholder} src={source}>
+					{selectionOverlay}
+				</Cell>
+				{subComponents}
+			</Column> : <Row {...rest} className={classes} inline>
+				<Cell className={css.image} component={ImageComponent} placeholder={placeholder} shrink src={source}>
+					{selectionOverlay}
+				</Cell>
+				<Cell align="center">
+					{subComponents}
+				</Cell>
+			</Row>
```

Comment:
How about simplifying render function by wrapping subComponents with Cell in `computed` block?

```
	computed: {
		...
		subComponents: ({caption, captionComponent: Caption, css, orientation, subCaption, subComponents}) => {
			const Component = (
				subComponents ? subComponents : <React.Fragment>
					{caption ? (<Cell className={css.caption} component={Caption} shrink>{caption}</Cell>) : null}
					{subCaption ? (<Cell className={css.subCaption} component={Caption} shrink>{subCaption}</Cell>) : null}
				</React.Fragment>
			);

			return orientation === 'horizontal' ? 
				<Cell align="center" component={Component} /> :
				component;
		}
	},

	render: (...) => {
		...
		const isHorizontal= orientation === 'horizontal';
		const Component = isHorizontal ? Row : Column;

		return (
			<Component {...rest} className={classes} inline>
				<Cell className={css.image} component={ImageComponent} placeholder={placeholder} shrin={isHorizontal} src={source}>
					{selectionOverlay}
				</Cell>
				{subComponents}
			</Component>
		);
	}
```

Reply:
I guess so... It looks simpler but doesn't seem good to read, though.